qiskit_experiments/test/pulse_backend.py

```diff
@@ -395,18 +421,28 @@ def run(self, run_input: Union[QuantumCircuit, List[QuantumCircuit]], **run_opti
             state_t = self.ground_state.copy()
             for instruction in circuit.data:
                 qubits, params, inst_name = self._get_info(circuit, instruction)
-                if inst_name in ["barrier", "measure"]:
+                if inst_name == "barrier":
+                    continue
+                if inst_name == "measure":
+                    meas_qubits += [qubits[0]]
                     continue
                 if inst_name == "rz":
+                    # TODO: Fix: Assumes 3 level system.
                     # Ensures that the action in the qubit space is preserved.
-                    unitary = np.diag([np.exp(1.0j * idx * params[0] / 2) for idx in [-1, 1, 3]])
+                    unitary_1q = np.diag([np.exp(1.0j * idx * params[0] / 2) for idx in [-1, 1, 3]])
+                    if qubits[0] == 0:
+                        unitary = np.kron(np.eye(3), unitary_1q)
+                    elif qubits[0] == 1:
+                        unitary = np.kron(unitary_1q, np.eye(3))
+                    else:
+                        raise QiskitError("Rz not implemented for more than 2 qubits")
```

Comment:
This seems like a piece of code that should be easy to generalize. E.g. something like this. Is there anything like this in the quantum info. module? I did not check.
```python
def _embed_single_element_gate(self, unitary: np.array, pos: int, n_elements: int) -> np.array:
    """Embed the gate of a single element in a larger Hilbert space.
    
    Args:
        pos: the position of the element in the larger Hilber space.
        n_elements: The number of individual elements in the Holbert space.
    """
    sub_gates = [np.eye(dim, dtype=complex)] * n_elements
    sub_gates[qidx] = unitary

    full_gate = None
    for gate in sub_gates:
        if full_gate is None:
            full_gate = gate
        else:
            full_gate = np.kron(gate, full_gate)

    return full_gate
```

Reply:
quantum_info does not seem to have anything for this. Nevertheless, why would we want to implement a general case? We don't need them for the tests, and if, at some point in the future if qiskit-experiments need this, it can be easily implemented. So, for now, we could keep it simple.

Reply:
added rz as subclass method